datatypes/src/raster/grid_traits.rs

```diff
@@ -146,9 +146,12 @@ pub trait NoDataValue {
 
     fn no_data_value(&self) -> Option<Self::NoDataType>;
 
+    #[allow(clippy::eq_op)]
     fn is_no_data(&self, value: Self::NoDataType) -> bool {
-        self.no_data_value()
-            .map_or(false, |no_data_value| value == no_data_value)
+        self.no_data_value().map_or(false, |no_data_value| {
+            // value is equal no-data value or both are NAN.
+            value == no_data_value || no_data_value != no_data_value && value != value
```

Comment:
1. Haven't we run into this problem already?

2. So if value != value but NO DATA is not NAN, then it is not NODATA?

3. Moreover, I guess it would help to put braces to this statement.

4. I guess we also should add a test to this function for checking that it works. It is pretty central.

Reply:
93be0ec